diff -N -u -r '--exclude=.git' '--exclude=*~' /home/lypo/genodelabs/goa_projects/pkcs11_tool/var/contrib/src/src/tools/pkcs15-tool.c src/src/tools/pkcs15-tool.c
--- /home/lypo/genodelabs/goa_projects/pkcs11_tool/var/contrib/src/src/tools/pkcs15-tool.c	2022-06-16 12:41:00.021857624 +0200
+++ src/src/tools/pkcs15-tool.c	2022-06-16 13:51:16.974456015 +0200
@@ -22,8 +22,6 @@
 
 #ifdef __APPLE__
 #define _XOPEN_SOURCE 600
-#else
-#define _XOPEN_SOURCE 500
 #endif
 #include <assert.h>
 #include <ctype.h>
